pkg/config/loader.go

```diff
@@ -59,6 +66,69 @@ func LoadResourceConfig() map[string]ResourceConfig {
 	return getDefaultResourceConfig()
 }
 
+func LoadResourceConfigFromConfigMap(vpcCniConfigMap *v1.ConfigMap) map[string]ResourceConfig {
+	resourceConfig := getDefaultResourceConfig()
+
+	warmIPTarget, minIPTarget, warmPrefixTarget := ParseWinPDTargets(vpcCniConfigMap)
+
+	// If no PD configuration is set in configMap or none is valid, return default resource config
+	if warmIPTarget == 0 && minIPTarget == 0 && warmPrefixTarget == 0 {
+		return resourceConfig
+	}
+
+	resourceConfig[ResourceNameIPAddressFromPrefix].WarmPoolConfig.WarmIPTarget = warmIPTarget
+	resourceConfig[ResourceNameIPAddressFromPrefix].WarmPoolConfig.MinIPTarget = minIPTarget
+	resourceConfig[ResourceNameIPAddressFromPrefix].WarmPoolConfig.WarmPrefixTarget = warmPrefixTarget
+
+	return resourceConfig
+}
+
+// ParseWinPDTargets parses config map for Windows prefix delegation configurations set by users
+func ParseWinPDTargets(vpcCniConfigMap *v1.ConfigMap) (warmIPTarget int, minIPTarget int, warmPrefixTarget int) {
+	var log = ctrl.Log.WithName("parse windows prefix delegation targets from config map")
```

Comment:
I think we better pass logger from `ParseWinPDTargets`'s callers instead of creating a new log instance. I saw only two callers, provider and ConfigMap_controller both which have logger you can pass here.

Reply:
Sounds good, I've updated accordingly.